src/core/heating_cooling_loads.jl

```diff
@@ -55,34 +60,34 @@ struct DomesticHotWaterLoad
         blended_doe_reference_percents::Array{<:Real,1} = Real[],
         annual_mmbtu::Union{Real, Nothing} = nothing,
         monthly_mmbtu::Array{<:Real,1} = Real[],
-        # addressable_load_fraction,  # TODO
+        addressable_load_fraction::Union{<:Real, AbstractVector{<:Real}} = 1.0,
         fuel_loads_mmbtu_per_hour::Array{<:Real,1} = Real[],
         time_steps_per_hour::Int = 1, # corresponding to `fuel_loads_mmbtu_per_hour`
-        latitude::Real=0.0,
-        longitude::Real=0.0
+        latitude::Real = 0.0,
+        longitude::Real = 0.0
     )
         if length(fuel_loads_mmbtu_per_hour) > 0
 
             if !(length(fuel_loads_mmbtu_per_hour) / time_steps_per_hour ≈ 8760)
                 throw(@error "Provided domestic hot water load does not match the time_steps_per_hour.")
             end
 
-            loads_kw = fuel_loads_mmbtu_per_hour .* (KWH_PER_MMBTU * EXISTING_BOILER_EFFICIENCY)
+            loads_kw = fuel_loads_mmbtu_per_hour .* (KWH_PER_MMBTU * EXISTING_BOILER_EFFICIENCY) .* addressable_load_fraction
```

Comment:
@Bill-Becker  you could consider adding an error check that `addressable_load_fraction` is either a scalar or of length equal to 12 (if using monthly loads) or the length of `fuel_loads_mmbtu_per_hour`. Otherwise, the code will error at these matrix multiplications if the lengths don't align. I don't think it's super high priority.

Reply:
Added these length-validation checks.